File: 29968040/python-57482691/minecraft/world/tile/Bush.java
package com.mojang.minecraft.world.tile;

import com.mojang.minecraft.world.World;
import com.mojang.minecraft.world.Tesselator;
import com.mojang.minecraft.phys.AABB;
import java.util.Random;

public class Bush extends Block {
	protected Bush(int id) {
		super(id);
		this.tex = 15;
	}

	public void tick(World world, int x, int y, int z, Random random) {
		int below = world.getBlock(x, y - 1, z);
		if ((!world.isLit(x, y, z))
				|| ((below != Block.dirt.id) && (below != Block.grass.id))) {
			world.setBlock(x, y, z, 0);
		}
	}
	@Override
	public void render(Tesselator t, World world, int layer, int x, int y, int z) {
		if ((world.isLit(x, y, z) ^ layer != 1))
			return;

		int tex = getTexture(15);
		float u0 = tex % 16 / 16.0F;
		float u1 = u0 + 0.0624375F;
		float v0 = tex / 16 / 16.0F;
		float v1 = v0 + 0.0624375F;

		int rots = 2;
		t.color(1.0F, 1.0F, 1.0F);
		for (int r = 0; r < rots; r++) {
			float xa = (float) (Math.sin(r * 3.141592653589793D / rots
					+ 0.7853981633974483D) * 0.5D);
			float za = (float) (Math.cos(r * 3.141592653589793D / rots
					+ 0.7853981633974483D) * 0.5D);
			float x0 = x + 0.5F - xa;
			float x1 = x + 0.5F + xa;
			float y0 = y + 0.0F;
			float y1 = y + 1.0F;
			float z0 = z + 0.5F - za;
			float z1 = z + 0.5F + za;

			t.vertexUV(x0, y1, z0, u1, v0);
			t.vertexUV(x1, y1, z1, u0, v0);
			t.vertexUV(x1, y0, z1, u0, v1);
			t.vertexUV(x0, y0, z0, u1, v1);

			t.vertexUV(x1, y1, z1, u0, v0);
			t.vertexUV(x0, y1, z0, u1, v0);
			t.vertexUV(x0, y0, z0, u1, v1);
			t.vertexUV(x1, y0, z1, u0, v1);
		}
	}

	public AABB getAABB(int x, int y, int z) {
		return null;
	}

	public boolean blocksLight() {
		return false;
	}

	public boolean isSolid() {
		return false;
	}
}

File: 29968040/python-57482691/minecraft/world/tile/Bush.py
from .Block import Block
import math


class Bush(Block):
    def __init__(self, _id: int):
        super().__init__(_id)
        self.tex = 15

    def tick(self, world, x: int, y: int, z: int, random):
        below: int = world.getBlock(x, y - 1, z)
        if not world.isLit(x, y, z) or (below != Block.dirt.id and below != Block.grass.id):
            world.setBlock(x, y, z, 0)

    def render(self, t, world, layer: int, x: int, y: int, z: int):
        if world.isLit(x, y, z) ^ layer != 1:
            return

        tex: int = self.getTexture(15)
        u0: float = tex % 16 / 16.0
        u1: float = u0 + 0.0624375
        v0: float = (tex // 16) / 16.0
        v1: float = v0 + 0.0624375

        rots: int = 2
        t.color(1.0, 1.0, 1.0)
        for r in range(rots):
            xa: float = math.sin(r * math.pi / rots + 0.7853981633974483) * 0.5
            za: float = math.cos(r * math.pi / rots + 0.7853981633974483) * 0.5
            x0: float = x + 0.5 - xa
            x1: float = x + 0.5 + xa
            y0: float = y + 0.0
            y1: float = y + 1.0
            z0: float = z + 0.5 - za
            z1: float = z + 0.5 + za

            t.vertexUV(x0, y1, z0, u1, v0)
            t.vertexUV(x1, y1, z1, u0, v0)
            t.vertexUV(x1, y0, z1, u0, v1)
            t.vertexUV(x0, y0, z0, u1, v1)

            t.vertexUV(x1, y1, z1, u0, v0)
            t.vertexUV(x0, y1, z0, u1, v0)
            t.vertexUV(x0, y0, z0, u1, v1)
            t.vertexUV(x1, y0, z1, u0, v1)

    def getAABB(self, x: int, y: int, z: int):
        return None

    def blocksLight(self) -> bool:
        return False

    def isSolid(self) -> bool:
        return False
